src/mlpack/core/tree/rectangle_tree/hilbert_r_tree_split_impl.hpp

```diff
@@ -1 +1,335 @@
+/**
+ * @file hilbert_r_tree_split_impl.hpp
+ * @author Mikhail Lozhnikov
+ *
+ * Implementation of class (HilbertRTreeSplit) to split a RectangleTree.
+ */
+#ifndef MLPACK_CORE_TREE_RECTANGLE_TREE_HILBERT_R_TREE_SPLIT_IMPL_HPP
+#define MLPACK_CORE_TREE_RECTANGLE_TREE_HILBERT_R_TREE_SPLIT_IMPL_HPP
 
+#include "hilbert_r_tree_split.hpp"
+#include "rectangle_tree.hpp"
+#include <mlpack/core/math/range.hpp>
+
+namespace mlpack {
+namespace tree {
+
+template<typename TreeType>
+void HilbertRTreeSplit::
+SplitLeafNode(TreeType *tree,std::vector<bool>& relevels)
+{
+  // If we are splitting the root node, we need will do things differently so
+  // that the constructor and other methods don't confuse the end user by giving
+  // an address of another node.
+  if (tree->Parent() == NULL)
+  {
+    // We actually want to copy this way.  Pointers and everything.
+    TreeType* copy = new TreeType(*tree, false);
+    copy->Parent() = tree;
+    tree->Count() = 0;
+    tree->NullifyData();
+    // Because this was a leaf node, numChildren must be 0.
+    tree->Children()[(tree->NumChildren())++] = copy;
+    HilbertRTreeSplit::SplitLeafNode(copy,relevels);
+    return;
+  }
+
+  TreeType *parent = tree->Parent();
+
+  size_t iTree = 0;
+  for(iTree = 0;parent->Children()[iTree] != tree; iTree++);
+
+  // Try to find splitOrder cooperating siblings in order to redistribute
+  // points among them and avoid split.
+  size_t firstSibling,lastSibling;
+  if(FindCooperatingSiblings(parent,iTree,firstSibling,lastSibling))
```

Comment:
Minor style issue, can you make this line:
```
if (FindCooperatingSiblings(parent, iTree, firstSibling, lastSibling))
```
(horizontal whitespace is what I'm focused on there)

I think this happens many other places, could you fix those too please?